devito/passes/iet/mpi.py

```diff
@@ -62,6 +62,23 @@ def _hoist_halospots(iet):
     Hoist HaloSpots from inner to outer Iterations where all data dependencies
     would be honored.
     """
+
+    # Hoisting rules -- if the retval is True, then it means the input `dep` is not
+    # a stopper to halo hoisting
+
+    def rule0(dep, candidates):
+        # E.g., `dep=W<f,[x]> -> R<f,[x-1]>` and `candidates=(time, x)` => False
+        # E.g., `dep=W<f,[t1, x, y]> -> R<f,[t0, x-1, y+1]>`, `dep.cause={t,time}` and
+        #       `candidates=(x,)` => True
+        return (all(d in dep.distance_mapper for d in candidates) and
```

Comment:
this line is the key fix, but to write it, I had to tweak/generalise (perhaps fix?) `TimedAccess.distance`